Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -836,21 +1094,43 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::join)
         s_array_join_seen_objects.remove(this_object);
     };
 
+    // 2. Let len be ? LengthOfArrayLike(O).
     auto length = TRY(length_of_array_like(vm, this_object));
-    DeprecatedString separator = ",";
-    if (!vm.argument(0).is_undefined())
-        separator = TRY(vm.argument(0).to_deprecated_string(vm));
+
+    String separator_string;
+
+    // 3. If separator is undefined, let sep be ",".
+    if (separator.is_undefined())
+        separator_string = ","_short_string;
+    // 4. Else, let sep be ? ToString(separator).
+    else
+        separator_string = TRY(separator.to_string(vm));
```

Comment:
Not sure if this is correct. A bit confused by the string/stringview/shortstring/deprecatedstring situation.

Reply:
This is correct, yep.